### PR TITLE
Add test case for shutdown

### PR DESCRIPTION
**Pull request by GuanLuo**

### CODE REVIEW

File: qa/L0_lifecycle/lifecycle_test.py [hunk 1]
```diff
@@ -2236,6 +2237,149 @@ def test_config_override(self):
...
+        while ((len(async_results) < request_count) and time_out > 0):
+            time_out = time_out - 1
+            time.sleep(1)
+
+        # Previous requests should succeed
+        for result in async_results:
+            if type(result) == InferenceServerException:
+                raise result
+            output_data = result.as_numpy('OUTPUT0')
+            self.assertTrue(np.allclose(output_data, input_data),
```
Comment (dyastremsky) [inline qa/L0_lifecycle/lifecycle_test.py:2293]: An optional tidbit: I know we use assertTrue and np.allclose together in most places, but another option is Numpy's built-in function for this. It addresses edge cases (that may not be relevant for Triton) and provides clearer output. Sharing below, feel free to ignore/resolve this comment.

https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html

File: qa/L0_lifecycle/test.sh [hunk 2]
```diff
@@ -1651,6 +1651,101 @@ wait $SERVER_PID
...
+fi
+set -e
+
+# check server log
+if [ `grep -c "custom_sequence_int32 v1: 2 in-flight inferences" $SERVER_LOG` == "0" ]; then
+    echo -e "\n***\n*** Expect logging for model and in-flight inference count\n***"
+    RET=1
+fi
+if [ `grep -c "custom_sequence_int32 v1: 1 in-flight inferences" $SERVER_LOG` == "0" ]; then
+    echo -e "\n***\n*** Expect logging for model and in-flight inference count\n***"
```
Comment (dyastremsky) [inline qa/L0_lifecycle/test.sh:1742]: May be worthwhile to differentiate this error message from the one above. It'd be hard to tell which fail from the message alone, so you'd have to check both logs.

**Review by dyastremsky — APPROVED** [review]

**Review by Tabrizian — COMMENTED** [review]
Do we need to also include a test case for ensemble?

**Comment by tanmayv25** [comment]
Looks good to me. I too think we must add testing for ensemble models.

**Review by dyastremsky — APPROVED** [review]

File: qa/L0_lifecycle/ensemble_zero_1_float32/config.pbtxt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        key: "INPUT0"
+        value: "INPUT0"
+      }
+      output_map {
+        key: "OUTPUT0"
+        value: "OUTPUT0"
+      }
+    }
+  ]
+}
```
Comment (tanmayv25) [inline qa/L0_lifecycle/ensemble_zero_1_float32/config.pbtxt:59]: newline

**Review by tanmayv25 — APPROVED** [review]
minor comment.

**Review by Tabrizian — APPROVED** [review]

**Pull request merged**